requests_toolbelt/adapters/source.py

```diff
@@ -28,7 +28,10 @@ class SourceAddressAdapter(HTTPAdapter):
         s.mount('http://', SourceAddressAdapter('10.10.10.10'))
     """
     def __init__(self, source_address, **kwargs):
-        self.source_address = source_address
+        if isinstance(source_address, basestring):
+            self.source_address = (source_address, 0)
+        else:
```

Comment:
This should be an `elif` to check that it's a tuple.

The final else should raise a `TypeError` to prevent any further confusingly "silent" error on our part.